corelibs/libgraph.md

```diff
@@ -1,36 +1,352 @@
+  - `GRAPH_DEFMODE` - default graphics mode,
+  - `GRAPH_ON` - display enabled mode,
+  - `GRAPH_OFF` - display disabled mode,
+  - `GRAPH_STANDBY` - display standby mode,
+  - `GRAPH_SUSPEND` - display suspend mode,
+  - `GRAPH_800x600x8` - 800x600 resolution, 8-bit color mode,
+  - `GRAPH_320x200x16` - 320x200 resolution, 16-bit color mode,
+  - `GRAPH_1920x1080x32` - 1920x1080 resolution, 32-bit color mode.
+
+  There should also be specified the _`freq`_ argument (`graph_freq_t` enum). There are few examples:
+  - `GRAPH_DEFFREQ` - default refresh rate,
+  - `GRAPH_24Hz` - 24Hz  refresh rate,
+  - `GRAPH_60Hz` - 60Hz  refresh rate,
+  - `GRAPH_120Hz` - 120Hz refresh rate,
+  - `GRAPH_360Hz` - 360Hz refresh rate.
+
+- `int graph_line(graph_t *graph, unsigned int x, unsigned int y, int dx, int dy, unsigned int stroke, unsigned int color, graph_queue_t queue)` </br>
+  Draws a line for the specified graphics adapter context. The following arguments should be passed:
+  - _`graph`_ - initialized `graph_t` structure,
+  - _`x`_ - start point x position, where 0 is the left edge of the screen,
+  - _`y`_ - start point y position, where 0 is the upper edge of the screen,
+  - _`dx`_ - relative end point x position,
+  - _`dy`_ - relative end point y position,
+  - _`stroke`_ - line thickness in pixels,
+  - _`color`_ - color,
+
+    - for `8-bit` color depth - the color palette is presented below. So `0x0B` represents cyan.
+
+      ![image](_images/def_color_palette.png)
+
+      Source: https://www.fountainware.com/EXPL/vga_color_palettes.htm
+
+    - for `16-bit` color depth - it's the following format in bits: `RRRRRGGGGGGBBBBB`, so `0x07E0` represents green.
+
+    - for `24-bit` color depth - it's represented by `0xRRGGBB` so `0x0000FF` is blue.
+
+    - for `32-bit` color depth - it's the following format in hex: `0xAARRGGBB`, where `A` represents alpha. With `AA` set to `0xFF` opacity equals 100%, but by default transparency isn't enabled so there can be any value.
+
+  - _`queue`_ - graphics task queue, should be chosen from the `graph_queue_t` enum, where are following options: `GRAPH_QUEUE_HIGH` - high priority queue, `GRAPH_QUEUE_LOW` - low priority queue, `GRAPH_QUEUE_BOTH` - both queues are chosen, `GRAPH_QUEUE_DEFAULT` - default queue.
+
+
+- `int graph_rect(graph_t *graph, unsigned int x, unsigned int y, unsigned int dx, unsigned int dy, unsigned int color, graph_queue_t queue)` </br>
+  Draws a rectangle. Arguments are similar to those in graph_line function. A drawn rectangle will be filled with a specified color.
+
+- `int graph_fill(graph_t *graph, unsigned int x, unsigned int y, unsigned int color, graph_fill_t type, graph_queue_t queue)` </br>
+  Fills polygon limited by drawn lines (when type is `GRAPH_FILL_FLOOD`), or limited by screen boundaries (`GRAPH_FILL_BOUND` type).
```

Comment:
Both `GRAPH_FILL_FLOOD` and `GRAPH_FILL_BOUND` fill insides of polygons with color ((`x`, `y`) should be any point inside the polygon to fill), but work differently:
- `GRAPH_FILL_FLOOD` works like windows paint bucket tool (floods homogeneous area - all pixels inside the polygon with color values same as the one at (`x`, `y`) flood origin point),
- `GRAPH_FILL_BOUND` fills the polygon until an edge of the same color as the fill color is found. It means that we cannot fill the polygon with the same color as the polygon boundary color,

Reply:
Thanks, I wasnt sure about that. I added example (will be visible after next force push), but I don't understand `It means that we cannot fill the polygon with the same color as the polygon boundary color,` sentence. Do you mean `can` instead of cannot, yes?

Reply:
Sorry for bad phrasing. I meant a polygon with differently colored edges.
Consider a rectangle with red top edge, green other edges and white fill color. If you'd try to fill the rectangle with red color than you can do it with `GRAPH_FILL_FLOOD ` method (all white pixels inside will become red). However, if you'd pass `GRAPH_FILL_BOUND` method than the whole screen will become red (the flooding will spill through the green edges).

Reply:
I hope new examples of `graph_fill()` usage, both for `GRAPH_FILL_FLOOD` and `GRAPH_FILL_BOUND` option explain properly how it works.